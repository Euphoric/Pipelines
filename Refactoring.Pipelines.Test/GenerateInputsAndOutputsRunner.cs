using System;
using System.Collections.Generic;
using System.Text;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using ApprovalTests;
using ApprovalTests.Core;
using ApprovalTests.Reporters;

namespace Refactoring.Pipelines.Test
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class GenerateInputsAndOutputsRunner
    {
        [TestMethod]
        public void GenerateInputsAndOutputs()
        {
            var inputCount = 2;
            var outputCount = 2;
            var subject = new InputsAndOutputsGenerator(inputCount, outputCount);
            Approvals.Verify(subject);
        }

        [TestMethod]
        public void GenerateInputsExtensions()
        {
            var inputCount = 2;
            var subject = new InputsExtensionsGenerator(inputCount);
            Approvals.Verify(subject);
        }
    }
}
