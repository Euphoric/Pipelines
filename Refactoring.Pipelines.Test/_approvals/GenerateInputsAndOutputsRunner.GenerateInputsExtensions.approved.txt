
namespace Refactoring.Pipelines.InputsAndOutputs
{
    public static class Inputs2Extensions
    {
        public static Inputs<TInput1, TInput2> GetInputs<TInput1, TInput2>(this IGraphNode node)
        {
            return new Inputs<TInput1, TInput2>(node);
        }
    }
}
